### ISSUE TITLE
Asset Groups and Subgroups

### ISSUE THREAD

**Issue by kuchedav** [dagster-io/dagster#11826]
### What's the use case?

The Dagster Asset graph can become very large. To better manage the workflows it would be useful to be able to define not noly categories but also sub-categories.

### Ideas of implementation

@dagster.asset(group_name=["group_name", "sub_group_name"])

Groups could be defined as a list (maybe with N levels).

On a visualization standpoint from the global asset graph.
1. Subgroups appear normally like groups do
2. Subgroups could be grouped to one box in the global asset graph.
3. Subgroups do not appear in the global asset graph but only when the Supergroup has been selected.

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by mkleinbort-ic** [comment]
I 100% would love this feature 

I use the dagit graph to communicate with my stakeholders, and a good clean UI is extremely valuable to drive trust and adoption

I asked about this in the slack some time ago, this was my mock-up of what I was asking for:

![image](https://user-images.githubusercontent.com/112868935/214889346-5c7bc7ba-12bf-4347-8d77-366de6b32f8e.png)

**Comment by mkleinbort-ic** [comment]
Also had the same syntax in mind:

```python
@dagster.asset(group_name=["group_name", "sub_group_name"])
def my_new_super_cool_asset(dep1, dep2, dep3):
    ...
```
----
Ps. I'm not sure how easy it would be to solve how to render the UI, in theory this could be very hard without validation of the "group" hierarchy.

Consider

```python
@asset(group_name = ['townspeaople', 'barbers'])
def X():
    ....

@asset(group_name = ['barbers', 'men'])
def Y():
    ....

@asset(group_name = ['men', 'townspeaople'])
def Z():
    ....
```

**Comment by mkleinbort-ic** [comment]
I'd accept the limitation that sub-groups MUST be wholly contained within a single group.

**Comment by kuchedav** [comment]
I agree, the groups should be more similar to a file-system than a tag-system. Hence in your example 'barbers' and 'men' would occur multiple times on different levels of hirarchy instead of being grouped together.

Also considering the case:
```
@asset(group_name = ['computer', 'mouse'])
def X():
    ....

@asset(group_name = ['mammal', 'mouse'])
def Y():
    ....
```
This would be two differnt sub-categories 'mouse' and would not be combined to one group.

**Comment by mkleinbort-ic** [comment]
I agree. It's similar to naming the groups

```python
@asset(group_name = 'computer:mouse')
def X():
    ....

@asset(group_name = 'mammal:mouse'])
def Y():
    ....
```
Then most of the logic is on the dagit UI. 

Do you have any estimate of when this would get done (if it gets done?) - I was close to clobbering together this feature myself, but I don't know any front-end coding.

**Comment by sryza** [comment]
Hi @mkleinbort-ic - this is a solid idea that we are likely interested in implementing. However, other parts of our roadmap are higher priority right now, so it's unlikely that we'll add this within the next couple months.

**Comment by mkleinbort-ic** [comment]
Thanks @sryza 

For the time being I am writing my assets like this:

```python
@asset(group_name = group_hirearchy(['computer', 'mouse']))
def X():
    ....

@asset(group_name = group_hirearchy(['mammal', 'mouse']))
def Y():
    ....
```

Where `group_hirearchy` is an often modified function 

```python
def group_hirearchy(hirearchy:list[str])->str|None:
    ...

```

That I can modify to change the layout of the Dagster UI/groups depending on my work that day.

For example:

```python
def group_hirearchy(hirearchy:list[str])->str|None:
    
    if 'FEATURE_22' in hirearchy:
        return '__'.join(hirearchy)
    else: 
        return None

```

**Comment by IAL32** [comment]
Our company is multi-tenant organization, and the only thing we can do for now is just assigning groups to each tenant's name.
We cannot do further subgrouping, which makes it very hard to navigate the UI...

**Comment by aschreiber1** [comment]
This would be very useful for the above reason, it seems that this is needed if a dagster would be a single control plane for an entire organization and not just a per team product

**Comment by quentin-sensome** [comment]
Looking forward a lot to that feature !

**Comment by alexcochran** [comment]
This would be very advantageous for us. We manage a large number of file-based feeds incoming from different vendors, some of which have multiple lines of business. Visually parsing a lineage with many assets could be much easier for stakeholders w/ subgrouping

**Comment by pbrady32** [comment]
Throwing in my vote for this as well. We are getting to the point where not having subgroups available as an organizational option is starting to cause some issues, and we are relatively small compared to what other folks have mentioned.

**Comment by pxlhuntr** [comment]
Another vote for this from me. @mkleinbort-ic I like that idea, thanks for sharing that. I'd still prefer having something out of box in any case!

**Comment by mscottnelson** [comment]
Are subgroups possible yet? I have a lot of related but independent pipelines that fan out through a deep tree, and grouping related nodes together would be so useful for communicating across stakeholders. I'm finding workarounds, but they go against the grain of the UI.

**Comment by alexcochran** [comment]
[An update to the changelog from 2026-06-11 mentions the availability of group hierarchies](https://github.com/dagster-io/dagster/blob/master/CHANGES.md#:~:text=1%2E13%2E9,graph):

> Asset group names may now contain `/` separators to define hierarchical groups (e.g. `marketing/ads`). Hierarchical groups can be selected with wildcards in asset selections (e.g. `group:"marketing/*"`) and are rendered as nested groups in the asset graph.

I think this meets the need for asset subgroups.

**Comment by krukowski** [comment]
Confirming this shipped in 1.13.9 (using a `/`  separator, not an array as initially proposed), and it's now referenced in [the docs](https://docs.dagster.io/guides/build/assets/metadata-and-tags/groups#nested-groups).
